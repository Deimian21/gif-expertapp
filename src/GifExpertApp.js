
import{ useState } from 'react';
import { AddCategory } from './Components/AddCategory';
import { GifGrid } from './Components/GifGrid';

const GifExpertApp = ({defaultCategories=[]}) =>{
  
   
   const [categories, setCategory] = useState([defaultCategories]);
  /*  const handleAdd = () =>{
   //setCategory ([...categories,'Comict']);
   setCategory( cats => [...categories,'Comics']);
   } */


   return (
<>
<h2> GifExpertApp</h2>
<AddCategory setCategory={setCategory}/>
<hr/>

   <ol>
      { categories.map(category =>(
       <GifGrid
       key={category} 
       category={category}
       />
      ))
   }
   </ol>
</>
)}
export default GifExpertApp;


