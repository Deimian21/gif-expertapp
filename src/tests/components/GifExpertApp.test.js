import { shallow }  from 'enzyme';
import React from 'react';
import GifExpertApp from '../../GifExpertApp';
describe('Pruebas en GifExpertApp', () => {
    test('debe contener el snapshot ', () => {
        const wrapper=shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
        
    });
    test('debe mosttar una lista de categorias', () => {
        const categories=['One Punch'];
        const wrapper=shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    })
    
})
