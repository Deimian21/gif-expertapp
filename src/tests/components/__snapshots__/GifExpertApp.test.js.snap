// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp debe contener el snapshot  1`] = `
<Fragment>
  <h2>
     GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category={Array []}
      key=""
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en GifExpertApp debe mosttar una lista de categorias 1`] = `
<Fragment>
  <h2>
     GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category={
        Array [
          "One Punch",
        ]
      }
      key="One Punch"
    />
  </ol>
</Fragment>
`;
